lib/doorkeeper/rails/helpers.rb

```diff
@@ -31,10 +31,16 @@ def doorkeeper_render_error
 
       def doorkeeper_render_error_with(error)
         options = doorkeeper_render_options(error) || {}
+        status = if options.delete(:not_found_for_forbidden) &&
+                    error.status == :forbidden
+                   :not_found
+                 else
+                   error.status
+                 end
         if options.blank?
-          head error.status
+          head status
         else
-          options[:status] = error.status
+          options[:status] = status
           options[:layout] = false if options[:layout].nil?
           render options
         end
```

Comment:
We can extract a method:

```ruby
def http_status_code
  if error.status == :forbidden && options[:respond_not_found_when_forbidden]
    :not_found
  else
    error.status
  end
end
```

And use it instead of `error.status` in this block, instead of adding the conditional in it.